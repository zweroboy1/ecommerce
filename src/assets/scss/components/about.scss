/* about */

.about {
  width: 100%;
  box-sizing: border-box;
  float: left;

  &__as {
    @include flex();

    @include media('m') {
      @include flexColumn();
    }

    p {
      @include font($size: $font-size-22);

      @include media('xs') {
        font-size: 14px;
      }
    }
  }

  &__rs-img {
    @include flexColumn();

    @include media('m') {
      @include flex($align: stretch);
      flex-direction: row;

      img {
        width: 100%;
        height: 100%;
        object-fit: cover;
        max-height: 200px;
      }
    }

    @include media('xs') {
      img {
        max-height: 120px;
      }
    }
  }

  &__team {
    // @include flex($wrap: wrap);
    @include flexColumn();
    gap: 50px;
  }

  &__member {
    @include flex($align: flex-start, $justify: flex-start);
    @include font($family: $font-family-title, $size: $font-size-28);
    flex: 1;
    width: 100%;
    gap: 50px;
    text-align: left;

    @include media('m') {
      flex-direction: column;
      align-items: center;
    }

    h2,
    p {
      text-align: left;
    }
  }

  &__photo {
    margin-bottom: 20px;
    border: 10px solid $button-second;
    box-shadow: 6px 7px 9px 5px #98969691;
    overflow: hidden;
    width: 420px;
    flex-shrink: 0;

    @include media('xs') {
      width: 300px;
    }
  }

  &__name {
    @include font($family: $font-family-title, $size: $font-size-28);
    margin: 0;
  }

  &__details {
    @include font($family: $font-family, $size: $font-size-22);

    p {
      font-size: 22px;
      padding: 15px 0 0;
      line-height: 1.4;
      text-align: justify;
    }

    span {
      @include font($family: $font-family-title, $size: $font-size-22);
    }

    ul li {
      font-size: 22px;

      &::before {
        content: 'âœ”';
        margin-right: 8px;
        color: $links;
        font-weight: bold;
      }
    }
  }

  .rs-link svg {
    background: url(/src/assets/svg/rs.svg) center / contain no-repeat;
    width: 6rem;
    height: 2rem;
    transition: $transition;

    &:hover {
      background: url(/src/assets/svg/rs-y.svg) center / contain no-repeat;
      width: 6.5rem;
      transform: scale(1.1);
      fill: $button-second;
    }
  }

  &__github {
    @include font($family: $font-family, $size: $font-size-16);
    color: $links;

    span {
      margin-right: 5px;
    }

    &:hover {
      transform: scale(1.05);
      color: $links-hover;
    }
  }

  .education {
    width: inherit;
    padding-left: 2%;
    padding-right: 2%;
    padding-bottom: 10px;

    &__item {
      position: relative;
      padding: 2em 0 0 0;
      width: 100%;
      margin: 0 auto;
    }

    &__item::before {
      content: '';
      position: absolute;
      top: 0;
      left: calc(13% - 1px);
      height: 100%;
      width: 2px;
      background: #4d4d4d;

      @include media('xs') {
        left: calc(15% - 1px);
      }
    }

    &__block:first-child {
      margin-top: 0;
    }

    &__block {
      position: relative;
      transition: all 0.3s ease;
      margin: 1em 0;
      display: flex;
    }

    &__content {
      left: 20%;
      position: relative;
      background: #4d4d4d;
      border-radius: 3px;
      margin-left: 0;
      padding: 0.5em;
      width: 74%;
      opacity: 0.9;

      @include media('xs') {
        left: 24%;
        width: 69%;
      }
    }

    &__content:hover {
      opacity: 1;
    }

    &__content:hover + .education__img {
      background-color: #b86f02;
    }

    &__content:before {
      content: '';
      position: absolute;
      top: 24px;
      right: 100%;
      height: 0;
      width: 0;
      border: 8px solid transparent;
      border-color: transparent;
      border-right-color: #4d4d4d;
    }

    &__title {
      color: #fafafa;
      margin: 0;
      padding: 2px 0;
      text-transform: uppercase;
      font-family: 'Montserrat-Bold';
      font-size: 0.9em;
    }

    &__speciality {
      margin: 0;
      padding: 4px 0;
      font-size: 0.9em;
      font-family: 'Montserrat-Bold';
    }

    &__p {
      color: #fbb03c;
      display: inline;
      font-size: 0.9em;
      transition: all 0.3s ease;
    }

    &__descr {
      color: #fafafa;
      font-size: 0.9em;
      line-height: 1.2;
      padding: 2px 0;
    }

    &__img {
      position: absolute;
      width: 35px;
      height: 35px;
      left: 9%;
      border-radius: 50%;
      background: #222;
      margin-top: 13px;
    }

    &__wrapp-img {
      display: block;
      width: 24px;
      height: 24px;
      position: relative;
      left: 50%;
      top: 50%;
      margin-left: -12px;
      margin-top: -12px;
      background-repeat: no-repeat;
      background-size: 100%;
    }

    &__date {
      display: inline-block;
      color: #4d4d4d;
      text-transform: uppercase;
      position: absolute;
      margin-left: 1%;
      font-size: 0.8em;
      left: auto;
      margin-top: 20px;
      display: block;
      width: 50px;

      @include media('xs') {
        font-size: 0.6em;
        width: 40px;
      }
    }

    &__block:after {
      display: table;
      clear: both;
      content: '';
    }
  }

  p,
  a {
    font-size: 22px;
  }
}
