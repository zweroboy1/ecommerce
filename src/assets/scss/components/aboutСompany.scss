/* about–°ompany */

.about-company {
  width: 100%;
  padding: 30px 20px;
  box-sizing: border-box;
  float: left;

  &__container {
    display: block;
  }

  &__title {
    @include flex($justify: flex-start, $align: flex-start, $wrap: nowrap);
    @include font($family: $font-family-title, $size: $font-size-24);
    border-radius: 12px 12px 0 0;
    margin-bottom: 10px;
  }

  &__body {
    position: relative;
    border-radius: 0 0 12px 12px;
  }

  &__content {
    @include font();
    max-width: 100%;
    word-wrap: break-word;

    p {
      padding: 0 0 15px;
      line-height: 1.4;
      text-align: justify;
    }
  }
}