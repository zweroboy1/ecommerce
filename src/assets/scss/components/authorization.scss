/* authorization */

.authorization {
  @include flex($justify: space-between, $align: baseline);
  width: 100%;
  padding: 20px;
  gap: 20px;

  @include media('m') {
    @include flexColumn($align: stretch);
    gap: 35px;
  }

  &__container,
  &__information {
    flex: 1;
    text-align: left;
  }

  &__container {

    .form-fields,
    .label {
      @include flexColumn($align: start);
      margin-bottom: 25px;
      position: relative;

      label {
        width: auto;
        float: left;
        color: $text-color;
        padding: 6px 0;
        margin-right: 10px;

        span {
          color: $important;
        }
      }

      input,
      select {
        width: 80%;
        min-width: 300px;

        @include media('xxs') {
          width: 75%;
          min-width: 200px;
        }
      }

      .password-block {
        width: 100%;

        button {
          background: transparent;
          font-size: 22px;
          padding: 0 1em;

          @include media('l') {
            padding: 0 0.3em;
          }

          &:focus,
          &:focus-visible,
          &:hover {
            border: none;
            outline: none;
          }
        }
      }
    }

    .label {
      span {
        width: auto;
        color: $text-color;
        float: left;
        padding: 6px 0;
        margin-right: 10px;
      }
    }

    .address-row {
      border: 1px solid $links;
      margin-bottom: 10px;
      padding: 10px;

      h2 {
        @include flex($align: flex-start, $wrap: wrap);

      }

      .label-check {
        display: block;
        margin: 15px 0;

        span {
          font-weight: bold;
        }
      }
    }
  }

  &__information {
    h4 {
      @include font($family: $font-family-title, $size: $font-size-19);
      padding-bottom: 10px;
      margin-bottom: 10px;
      color: #bdc3c7;
    }

    p {
      padding: 15px 0;
    }

    ul {
      @include media('m') {
        padding: 0 15px;
      }
    }

    li {
      padding: 5px 0;
      list-style: disc;
    }
  }

  .error-message,
  .email__error,
  .password__error {
    position: absolute;
    top: 70px;
    color: $important;
  }

  .error input,
  .error select,
  .error input:focus,
  .error select:focus,
  .error input:hover,
  .error select:hover,
  .input__error:focus,
  .input__error:hover,
  .input__error {
    color: $important;
    border-color: $important;
    box-shadow: 0 0 0 2px rgb(244 67 54 / 8%);
  }

  .service__error {
    @include font($family: $font-family-title, $size: $font-size-24);
    color: $error;
    text-align: center;
    margin-top: 25px;
  }
}