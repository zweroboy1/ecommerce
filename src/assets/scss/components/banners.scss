/* banners */

.banner {
  touch-action: manipulation;
  user-select: none;

  &__bg {
    max-width: 100%;
    height: 550px;
    position: relative;
    background-repeat: no-repeat;
    background-position: center;
    overflow: hidden;
    background: cover;
  }

  &__video {
    position: absolute;
    min-height: 100%;
    min-width: 100%;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    z-index: -1;
  }

  &__content {
    justify-content: center;
    flex-wrap: nowrap;
    flex-direction: column;
    align-items: center;
    height: 550px;
    max-width: inherit;
    padding: 0 calc(100% / 2 - 680px);
    display: flex;

    @media (max-width: 1023px) {
      padding: 0 15px;
    }
  }

  &__img {
    width: 2000px;
    height: 509px;
    flex: 1 100%;
    margin: 0 auto;
    justify-content: center;
    width: 100%;
    align-items: center;
    display: flex;
    overflow: hidden;
  }

  &__box {
    padding: 50px 0;
  }

  &__title {
    font-size: 32px;
    color: #000000;
  }

  &__subtitle {
    color: #000000;
    background: rgba(255, 255, 255, 70%);
    border-radius: 10px;
    padding: 7px 15px;
  }

  &__span {
    font-weight: 100;
    font-family: sans-serif;
    font-size: 32px;
  }

  &__txt {
    font-size: 18px;
    margin-top: 10px;
    line-height: 1.4;
    color: #000;
  }
}
