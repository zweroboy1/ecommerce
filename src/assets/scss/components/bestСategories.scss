/* best–°ategories */

.best-categories {
  width: 100%;
  margin-top: 30px;

  &__grid {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
  }

  &__block {
    display: flex;
    flex-direction: column;
    gap: 15px;
    flex: 0 0 auto;
    max-width: 350px;
    margin: 5px;
    padding-left: 0;
    padding-right: 0;
    box-sizing: border-box;
    float: left;
    min-height: 1px;
    margin-left: 10px;
  }

  &__item {
    overflow: hidden;
    border-radius: 20px !important;
  }

  &__bg {
    background-color: #f5f5f7;
    background-size: cover;
    height: 250px;
  }

  &__content {
    height: 250px;
  }

  &__img {
    padding: 20px 20px 0;
  }

  &__pict {
    max-height: 100%;
    width: auto;
    flex: 0;
    width: 600px;
    height: 462px;
  }

  &__box {
    padding: 20px;
  }

  &__title {
    font-size: 16px;
  }
}