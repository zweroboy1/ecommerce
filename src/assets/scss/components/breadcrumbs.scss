.breadcrumbs {
  position: relative;
  width: 100%;
  padding: 5px 0;
  margin-bottom: 14px;
  border-bottom: $border;
  text-align: left;

  @include media('xs') {
    padding: 0 10px;
  }

  &__link {
    display: inline-block;
    padding: 0;
    color: $top-links;
    line-height: 30px;
    height: 30px;
    font-size: $font-size-14;
  }

  &__slash {
    display: inline-flex;
    vertical-align: bottom;
    align-items: center;
    padding: 0;
    color: #737373;
    height: 30px;
    margin: 0px 6px;
  }

  &__current {
    display: inline-block;
    line-height: 30px;
    height: 30px;
    color: $important;
    text-decoration: double;
    padding: 0;
  }
}