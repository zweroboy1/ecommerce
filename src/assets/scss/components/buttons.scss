.button {
  background: $links;
  color: $background;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: max-content;
  min-width: $button-size-40;
  height: $button-size-40;
  white-space: normal;
  line-height: 1.2;
  padding: 5px 15px;
  border: none;
  background-clip: border-box;
  font-size: 15px;
  letter-spacing: normal;
  text-transform: none;
  border-radius: 40px;
  transition: $transition;

  &:hover {
    color: $background;
    background: $button-hover;
    box-shadow: inset 0 -2px 0 0 rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
}

.button-second {
  background: $button-second;
  color: $text-color;
  transition: $transition;

  &:hover {
    color: $text-color;
    background: $button-second-hover;
  }

  &.confirm {
    min-width: 161px;

    span {
      width: 100%;
      height: 100%;
      display: flex;
      justify-content: center;
      align-items: center;
      position: relative;
    }
  }
}

.button-order {
  &.confirm {
    min-width: 161px;

    span {
      width: 100%;
      height: 100%;
      display: flex;
      justify-content: center;
      align-items: center;
      position: relative;
    }
  }
}

.header-icon {
  display: block;
  position: relative;
  height: 35px;
  width: 35px;
  margin: 0;
  left: 0;
  top: -5px;
  right: 0;
  cursor: pointer;
  transition: $transition;

  &:hover {
    transform: scale(1.1);
  }

  @include media('xs') {
    height: 25px;
    width: 25px;
  }

  @include media('xxs') {
    height: 15px;
    width: 15px;
  }
}

.top-arrow {
  @include arrow();
}

.close {
  position: absolute;
  width: 25px;
  height: 25px;
  right: 10px;
  top: 10px;
  opacity: 0.75;
  color: #595959;
  cursor: pointer;
  transition: $transition;
  z-index: 10;

  span {
    position: absolute;
    width: 25px;
    height: 2px;
    left: 50%;
    top: 50%;
    background-color: #595959;
    transition: $transition;
  }

  span:nth-of-type(1) {
    transform: translate(-50%, 0%) rotate(45deg);
  }

  span:nth-of-type(2) {
    transform: translate(-50%, 0%) rotate(-45deg);
  }
}

.reset {
  position: relative;
  border: 1px solid #dd4b3926;
  color: #dd4b39;
  background: #dd4b390d;
  font-weight: 500;
  padding-right: 28px;
  padding: 7px 28px 7px 10px;
  margin: 4px 5px 4px 0;
  border-radius: $border-radius-8;
  transition: $transition;

  &:hover {
    border-color: #dd4b39;
    color: #dd4b39;
    cursor: pointer;
  }

  &__icon {
    content: '';
    position: absolute;
    width: 15px;
    height: 15px;
    position: absolute;
    top: 50%;
    right: 7px;
    transform: translateY(-50%);
    line-height: 100%;
    background: url(/src/assets/svg/reset.svg) no-repeat center center/cover;
  }
}

.compared-products__icon {
  background: url(/src/assets/svg/stats.svg) no-repeat center center / cover;
}

.wishlist__icon {
  background: url(/src/assets/svg/heart.svg) no-repeat center center / cover;
}

.my-account__icon {
  background: url(/src/assets/svg/profile.svg) no-repeat center center / cover;

  &:hover {
    filter: invert(100%) hue-rotate(8deg) brightness(162%);
  }
}

.minicart__icon {
  background: url(/src/assets/svg/cart.svg) no-repeat center center / cover;
}
