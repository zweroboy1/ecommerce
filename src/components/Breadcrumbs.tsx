const Breadcrumbs = () => {
  return (
    <div className="breadcrumbs">
      <a href="https://unitheme.net/" className="breadcrumbs__link">
        <bdi>Главная</bdi>
      </a>
      <span className="breadcrumbs__slash">/</span>
      <a href="https://unitheme.net/electronics/" className="breadcrumbs__link">
        <bdi>Электроника</bdi>
      </a>
      <span className="breadcrumbs__slash">/</span>
      <span className="breadcrumbs__current">
        <bdi>Телефоны</bdi>
      </span>
    </div>
  );
};

export { Breadcrumbs };
