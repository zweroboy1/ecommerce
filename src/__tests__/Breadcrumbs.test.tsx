import { render, screen } from '@testing-library/react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { Breadcrumbs } from '../components/Breadcrumbs';

describe('Breadcrumbs component', () => {
  it('displays home link when no breadcrumbs provided', () => {
    render(
      <MemoryRouter>
        <Breadcrumbs breadcrumbs={[]} />
      </MemoryRouter>
    );

    expect(screen.getByText('Главная')).toBeInTheDocument();
  });

  it('displays a single breadcrumb', () => {
    const { container } = render(
      <BrowserRouter>
        <Breadcrumbs breadcrumbs={[{ id: 1, name: 'Категория', url: 'category' }]} />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it('displays multiple breadcrumbs', () => {
    const { container } = render(
      <BrowserRouter>
        <Breadcrumbs
          breadcrumbs={[
            { id: 1, name: 'Категория', url: 'category' },
            { id: 2, name: 'Товар', url: 'product' },
          ]}
        />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it('generates correct links', () => {
    render(
      <BrowserRouter>
        <Breadcrumbs
          breadcrumbs={[
            { id: 1, name: 'Категория', url: 'category' },
            { id: 2, name: 'Товар', url: 'product' },
          ]}
        />
      </BrowserRouter>
    );

    expect(screen.getByText('Главная')).toBeInTheDocument();
    expect(screen.getByText('Категория')).toBeInTheDocument();
    expect(screen.getByText('Товар')).toBeInTheDocument();
  });
});
