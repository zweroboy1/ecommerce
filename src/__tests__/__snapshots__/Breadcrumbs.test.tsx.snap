// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component displays a single breadcrumb 1`] = `
<div>
  <div
    class="breadcrumbs"
  >
    <a
      aria-current="page"
      class="breadcrumbs__link active"
      href="/"
    >
      Главная
    </a>
    <span
      class="breadcrumbs__slash"
    >
      /
    </span>
    <span>
      <a
        aria-current="page"
        class="breadcrumbs__current active"
        href="/"
      >
        Категория
      </a>
    </span>
  </div>
</div>
`;

exports[`Breadcrumbs component displays multiple breadcrumbs 1`] = `
<div>
  <div
    class="breadcrumbs"
  >
    <a
      aria-current="page"
      class="breadcrumbs__link active"
      href="/"
    >
      Главная
    </a>
    <span
      class="breadcrumbs__slash"
    >
      /
    </span>
    <span>
      <a
        class="breadcrumbs__link"
        href="/catalog/category"
      >
        Категория
      </a>
    </span>
    <span>
      <span
        class="breadcrumbs__slash"
      >
        /
      </span>
      <a
        aria-current="page"
        class="breadcrumbs__current active"
        href="/"
      >
        Товар
      </a>
    </span>
  </div>
</div>
`;
